scripts/jest/matchers/__tests__/toWarnDev-test.ts

```diff
@@ -0,0 +1,50 @@
+/* tslint:disable:no-console */
+
+describe('toWarnDev', () => {
+  describe('usage', () => {
+    test('fails with incorrect type of message', () => {
+      expect(() => {
+        // @ts-ignore:next-line
+        expect(() => {}).toWarnDev(false);
+      }).toThrowErrorMatchingInlineSnapshot(
+        `"toWarnDev() requires a parameter of type string but was given boolean."`
+      );
+    });
+  });
+
+  if (__DEV__) {
+    describe('without message', () => {
+      test('does not fail if called', () => {
+        expect(() => {
+          expect(() => {
+            console.warn('warning');
+          }).toWarnDev();
+        }).not.toThrow();
+      });
+
+      test('fails if not called', () => {
+        expect(() => {
+          expect(() => {}).toWarnDev();
```

Comment:
here is another that might need to be ignored if you indeed need `// ts-ignore`

Reply:
See [comment above](https://github.com/algolia/instantsearch.js/pull/3574/files#r262070058).